textile/features.textile

```diff
@@ -419,6 +420,25 @@ h3(#rest-encryption). Encryption
 ** @(RSE2a)@ Takes an optional @keyLength@ parameter, which is the length in bits of the key to be generated. If unspecified, this is equal to the default @keyLength@ of the default algorithm: for @AES@, 256 bits.
 ** @(RSE2b)@ Returns (or calls back with, if the language cryptographic randomness primitives are blocking or async) the key as a binary (e.g. a byte array, depending on the language)
 
+h3(#rest-annotations). RestAnnotations
+
+* @(RSAN1)@ @RestAnnotations#publish@ function:
+** @(RSAN1a)@ Takes a first argument of a @Message@ identifier in the form of the @messageSerial@ string of the message to be annotated, and a second argument representing the annotation to be published.
+*** @(RSAN1a1)@ In languages where method overloading is supported, the first argument should also accept a @Message@ object (which must contain a populated @serial@ field)
+*** @(RSAN1a2)@ The form of the second argument may accept any language-idiomatic representation (e.g. plain objects in untyped languages), but must allow the user to supply at least the @type@, @clientId@, @name@, @count@, @data@, and @extras@ fields
+*** @(RSAN1a3)@ The SDK must validate that the user supplied a @TAN5b@ @type@. (All other fields are optional)
+** @(RSAN1c)@ The SDK must POST a request body consisting of an array with one element, which is the annotation to be published
+*** @(RSAN1c1)@ The @Annotation.action@ must be set to @ANNOTATION_CREATE@
+*** @(RSAN1c2)@ If the user has supplied an @Annotation.data@, that must be encoded (and the @encoding@ set) just as it would be for a @Message@, per @RSL4@
```

Comment:
Should we support library-generated idempotency keys for REST annotation publishing the same way we do for messages (RSL1k1)?

Reply:
Sure, good idea, added